pyEpiabm/pyEpiabm/tests/test_routine/test_toy_population_config.py

```diff
@@ -200,6 +205,16 @@ def test_assign_cell_locations_grid(self, cell_num):
                                    (i % grid_len) / (grid_len - 1))
             self.assertAlmostEqual(cell.location[1],
                                    (i // grid_len) / (grid_len - 1))
+            mcell_len = math.ceil(math.sqrt(len(cell.microcells)))
+            for j, mcell in enumerate(cell.microcells):
+                test_x = (cell.location[0] +
+                          (j % mcell_len - .5 * (mcell_len - 1)) /
+                          (grid_len * (mcell_len - 1)))
+                test_y = (cell.location[1] +
+                          (j // mcell_len - .5 * (mcell_len - 1)) /
+                          (grid_len * (mcell_len - 1)))
+                self.assertAlmostEqual(mcell.location[0], test_x)
+                self.assertAlmostEqual(mcell.location[1], test_y)
 
     def test_assign_cell_locations_known_grid(self):
```

Comment:
Might be nice to have a cheeky microcell position check in here too?
And also a check for the random method that all microcells belong to the nearest cell (more thought but definitely good to have!)